daemons/fenced/fenced_commands.c

```diff
@@ -2627,74 +2689,17 @@ st_child_done(int pid, const pcmk__action_result_t *result, void *user_data)
         cmd->device_next = cmd->device_next->next;
     }
 
-    /* this operation requires more fencing, hooray! */
-    if (next_device) {
-        log_async_result(cmd, result, pid, next_device->id, false);
-        schedule_stonith_command(cmd, next_device);
-        /* Prevent cmd from being freed */
-        cmd = NULL;
-        goto done;
-    }
-
-    send_async_reply(cmd, result, pid, false);
-
-    if (!pcmk__result_ok(result)) {
-        goto done;
-    }
-
-    /* Check to see if any operations are scheduled to do the exact
-     * same thing that just completed.  If so, rather than
-     * performing the same fencing operation twice, return the result
-     * of this operation for all pending commands it matches. */
-    for (gIter = cmd_list; gIter != NULL; gIter = gIterNext) {
-        async_command_t *cmd_other = gIter->data;
-
-        gIterNext = gIter->next;
-
-        if (cmd == cmd_other) {
-            continue;
+    if (next_device == NULL) {
+        send_async_reply(cmd, result, pid, false);
+        if (pcmk__result_ok(result)) {
+            reply_to_duplicates(cmd, result, pid);
         }
+        free_async_command(cmd);
```

Comment:
guess this should be moved to the end after the next_device != NULL case.

Reply:
I don't think so, in that case more fencing is being scheduled for the same cmd, so it should stay in cmd_list

Reply:
You're right - overlooked the 'cmd = NULL' above